src/context/playlist-context.js

```diff
@@ -1,10 +1,57 @@
-import { createContext, useContext, useReducer } from "react";
+import { createContext, useContext, useReducer, useEffect, useState } from "react";
 import { playlistReducer } from "../reducer";
 
 const PlaylistContext = createContext();
 
 const PlaylistProvider = ({ children }) => {
   const [{likedVideos, watchLater, history, option}, playlistDispatch] = useReducer(playlistReducer, { likedVideos: [], watchLater: [], history: [], option: "" });
+
+  useEffect(() => {
+    if (JSON.parse(localStorage.getItem("likes" ?? ""))){
+      const likes = JSON.parse(localStorage.getItem("likes"));
+      likes && likes.length > 0 && likes.map(video => playlistDispatch({
+        type: "LIKED",
+        payload: video
+      }));
+    }
+  }, [])
+
+  useEffect(() => {
+    if (JSON.parse(localStorage.getItem("watchlater" ?? ""))){
+      const watchlater = JSON.parse(localStorage.getItem("watchlater" ?? ""));
+      watchlater && watchlater.length > 0 && watchlater.map(video => playlistDispatch({
+        type: "WATCH_LATER",
+        payload: video
+      }));
+    }
+  }, [])
+
+  useEffect(() => {
+    if (JSON.parse(localStorage.getItem("history" ?? ""))){
+      const history = JSON.parse(localStorage.getItem("history" ?? ""));
+      history && history.length > 0 && history.map(video => playlistDispatch({
+        type: "HISTORY",
+        payload: video
+      }));
+    }
+  }, [])
+
+  useEffect(() => {
+    if (localStorage.getItem("option" ?? "")){
```

Comment:
`??` is not required here as null or empty string both result to falsy value. I dont understand the purpose of using nullish coalescing operator here

Reply:
I was trying to handle an edge case, but now I have made the changes and I think with nullish coalescing empty string is not a falsy value